tests/test_nn.py

```diff
@@ -524,6 +524,29 @@ def test_convtranspose3d(getkey):
     assert jnp.all(conv(data) == answer)
 
 
+def test_dot_product_attention_weights(getkey):
+    q = jnp.array([[0.0, 2**0.5]])
+    k = jnp.array([[1.0, 0.0], [0.0, 1.0]])
+    weights = eqx.nn.attention.dot_product_attention_weights(q, k)
```

Comment:
Can these new functions be added to `nn/__init__.py`, so that the lookup here can be `eqx.nn.dot_product_attention_weights`? (It's not really communicated at the moment but individual modules should be treated as private API.)

Reply:
Likewise, give these functions a docstring and add them to the documentation?

Alternatively, if you prefer, we can treat these as "semi-public" -- there for power users only, not necessarily with full docs etc. In which case let's expose these under `eqx.experimental`.

Reply:
I like the "semi-public" approach to start with -- It's the more conservative approach that lets us to make api changes if there are any oversights.

Reply:
Does this mean just add the imports in `eqx/experimental/__init__.py`?

Reply:
Actually what makes sense to me is to just keep the namespace in the private attention module -- and so users can use at their own risk until it does get added to the public api

Reply:
Sounds good to me. I can see that you were the one to originally open #142 so that'll suffice for now until someone else wants the same functionality.